components/channel_header_dropdown/menu_items/leave_channel/index.ts

```diff
@@ -1,17 +1,34 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {bindActionCreators, Dispatch} from 'redux';
-import {connect} from 'react-redux';
+import {bindActionCreators, Dispatch, ActionCreatorsMapObject} from 'redux';
+import {connect, ConnectedProps} from 'react-redux';
 
-import {GenericAction} from 'mattermost-redux/types/actions';
+import {GenericAction, Action} from 'mattermost-redux/types/actions';
+
+import {ModalData} from 'types/actions';
 
 import {leaveChannel} from 'actions/views/channel';
+import {openModal} from 'actions/views/modals';
 
 import LeaveChannel from './leave_channel';
 
-const mapDispatchToProps = (dispatch: Dispatch<GenericAction>) => ({
-    actions: bindActionCreators({leaveChannel}, dispatch),
-});
+type Actions = {
```

Comment:
I don't think we need to decide on the PropsFromRedux pattern in this PR, but this case does make that look a bit worse since we still have to define the types here to satisfy the typing for `bindActionCreators`. That might be resolved when we figure out the other redux typing issues though.

No changes to make here, but I wanted to mention that.

Reply:
yes as we observed that bindActionCreator is propagating correct types i removed the type `Actions` from here.
<img width="1172" alt="image" src="https://user-images.githubusercontent.com/17708702/140914310-6a5d59ab-3f3e-4b25-85e3-62a4ae78cdd8.png">

is it better ?